src/discussions/posts/post-editor/PostEditor.jsx

```diff
@@ -77,19 +79,30 @@ function PostEditor({
   } = useSelector(selectAnonymousPostingConfig);
   const cohorts = useSelector(selectCourseCohorts);
   const post = useSelector(selectThread(postId));
-  const initialValues = {
-    postType: post?.type || 'discussion',
-    topic: post?.topicId || topicId || nonCoursewareTopics?.[0]?.id,
-    title: post?.title || '',
-    comment: post?.rawBody || '',
-    follow: post?.following ?? true,
-    anonymous: allowAnonymous ? false : undefined,
-    anonymousToPeers: allowAnonymousToPeers ? false : undefined,
+  let initialValues = {
+    postType: 'discussion',
+    topic: topicId || nonCoursewareTopics?.[0]?.id,
+    title: '',
+    comment: '',
+    follow: true,
+    anonymous: false,
+    anonymousToPeers: false,
   };
+  if (editExisting) {
+    initialValues = {
+      postType: post.type || 'discussion',
+      topic: post.topicId || topicId || nonCoursewareTopics?.[0]?.id,
+      title: post.title || '',
+      comment: post.rawBody || '',
+      follow: (post.following === null || post.following === undefined) ? true : post.following,
```

Comment:
Not sure why we need to check for both `null` and `undefined`. I understand that we can't do `!post.following`, because it can be `false`, and we are trying to [catch "nullish" values](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator). Would that be better, since it's a more concise equivalent:
```suggestion
      follow: post.following ?? true,
```

Reply:
As I mentioned elsewhere currently ?? is breaking tests and lint checks in some cases, which is why I've switched it out. 

In other cases ?? is equivalent to || but not here, so I need to check for null or undefined.